### PR TITLE
replaced die() with an header for login page redirection

### PR DESCRIPTION
**Pull request by MigDinny**
#946 

die() was triggered with a message saying that basically a login was required. I replaced it with a header to redirect the browser to login-page

### CODE REVIEW

**Comment by MigDinny** [comment]
@aethelwulffe Can I create a dialog message with that information? I mean, we got redirected to login page and a dialog opens up telling us that we need to log-in first.

**Comment by yehster** [comment]
The die() in the original code prevents everything else in the script from running. Redirect with headers is ok, but cannot replace the die().  Otherwise this introduces a possible security risk.

**Comment by aethelwulffe** [comment]
It must die().

**Comment by MigDinny** [comment]
can it be header() + die()?

**Comment by yehster** [comment]
Yes, header + die() would be better, the the URL for the redirect needs to pass the current error message still (as Art mentioned before).  I'm not sure if the login page itself can handle that right now, so there may need to be additional changes to the login.php page as well.

**Comment by MigDinny** [comment]
@yehster It doesn't handle those messages, but I can do that. I was just checking with you two if there is a possibility of creating that error message. I'll work on that.

File: interface/login/login.php [hunk 1]
```diff
@@ -150,6 +150,14 @@ function imsubmitted() {
                                 </tr>
                             <?php endif; ?>
 
+                            <?php if (isset($_GET['loginfirst'])): ?>
+                                <tr>
+                                    <td colspan='2' class='text' style='color:red'>
+                                        <?php echo xlt('A login is required to access this platform.'); ?>
```
Comment (aethelwulffe) [inline interface/login/login.php]: This is fine (testing by setting session timeout to 5 minutes), but I would prefer some sort of explicit statement like "You have been logged out due to inactivity, or you have experienced an error.  Please re-enter access credentials."
The "Site ID is missing from session data!" message used to appear in a single frame when the application was open but attempted an action in that frame (frames are now iframe tabs).  This message was not explicit enough, and created confusion.  Booting to the gateway page is much better in some ways, *but if someone was working in a text field in another frame and got booted, they will lose their unsaved data without being able to copy it to the clipboard.*  Having a modal window re-login of some kind would be very helpful in this case, especially when it is an actual server error like mysql going poof.

My arguments above may not be totally coherent, but if we can moderate 1. confusion as to why they were booted out of the system (inactivity), 2. Loss of unsaved data, then that would be the preferred approach.

**Comment by aethelwulffe** [comment]
My change request was meant to be a comment.  If others are fine with the wording, then I am game.  This is a good thing here.  I just think we should make sure we do this in the best way for the users.  Even as-is, this is better than just a confusing message, with the exception that they instantly lose unsaved data.

**Comment by MigDinny** [comment]
@aethelwulffe honestly I'm super confused... Is any change needed for this PR except the message change? (I will send a commit to change the message to the one you've said)

**Comment by aethelwulffe** [comment]
You don't have to use my phrase word for word.  We just want to tell the users (in plain speech) WHY they got booted.

**Comment by aethelwulffe** [comment]
Yeah, just changing the phrase would please me enough.  All the same, this PR actually has a pretty significant effect in how the application works, so I would like to discuss the effects before merging.

**Comment by MigDinny** [comment]
@aethelwulffe i'm not creative enough, i've used your phrase haha

didn't have any better idea, check it out now

**Comment by aethelwulffe** [comment]
Three-day bump.
Please review.

**Comment by aethelwulffe** [comment]
This is ready for a second review.  Please, someone...

**Pull request merged**